### PR TITLE
fix generic struct2MapAll

### LINKED ISSUE #801: go-client 泛化调用go-server
Time.Time类型在传输过程中变成了nil,无论是是从client请求到server还是从server返回到client

另外自定义了重命名 Time.Time MyTime,增加序列化,情况依然没有变化

2020/10/22 21:22:24 dubbo client: ManageProvider Login [{admin 123456}]
2020/10/22 21:22:24 dubbo resp: map[createdAt:<nil> deletedAt:<nil> email:yangxiaolei@kuaifazs.com id:1 password:250415709c2a1e522232f86a7dbe553b salt:5Id0Ag state:1 token:eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwidXNlcl9uYW1lIjoiYWRtaW4iLCJlbWFpbCI6Inlhbmd4aWFvbGVpQGt1YWlmYXpzLmNvbSIsImJ1c2luZXNzIjozLCJleHAiOjE2MDMzODM3NDQsImlzcyI6ImFkc2Vuc2UtbWFuYWdlciJ9.3VG6LIENfPIogg5a9Fzy5fB3Ag0w_RXfGC0LCGiHfu4 updatedAt:<nil> userName:admin] <nil>

[2020-10-22/21:51:25 kuaifa.com/adsense/adsense/server/advertiser.(*AdvertiserProvider).Register: advertiser.go: 20] req:advertiser.Advertiser{Id:0, ScreenName:"杀猪的", Mobile:"18600363056", Qq:"12232", Wechat:"1111", AccountType:1, Balance:0, CertificatesImg:"11", BusinessScope:"", Status:0, Adjust:2, Sfrom:1, CreatedAt:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, UpdatedAt:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, Token:""}

另外咨询个问题,泛化调用的时候能不能声明返回值类型

### PR DESCRIPTION
**Pull request by fangyincheng**
**What this PR does**:

fix generic struct2MapAll

**Which issue(s) this PR fixes**:

Fixes #801

### CODE REVIEW

**Review by zouyx — COMMENTED** [review]
Check fail bro

File: filter/filter_impl/generic_filter.go [hunk 1]
```diff
@@ -93,11 +93,21 @@ func struct2MapAll(obj interface{}) interface{} {
 	if t.Kind() == reflect.Struct {
 		result := make(map[string]interface{}, t.NumField())
 		for i := 0; i < t.NumField(); i++ {
-			if v.Field(i).Kind() == reflect.Struct || v.Field(i).Kind() == reflect.Slice || v.Field(i).Kind() == reflect.Map {
+			switch v.Field(i).Kind() {
+			case reflect.Struct:
+				fallthrough
```
Comment (zouyx) [inline filter/filter_impl/generic_filter.go:98]: 骚气蓬勃的用法，方哥。

**Review by zouyx — APPROVED** [review]

File: filter/filter_impl/generic_filter.go (hunk 1 above)
Comment (AlexStocks) [reply filter/filter_impl/generic_filter.go:98]: 用法没问题，但是会降低执行效率

**Review by AlexStocks — APPROVED** [review]

File: filter/filter_impl/generic_filter.go [hunk 2]
```diff
@@ -93,13 +93,20 @@ func struct2MapAll(obj interface{}) interface{} {
...
-				if v.Field(i).CanInterface() {
-					setInMap(result, t.Field(i), struct2MapAll(v.Field(i).Interface()))
+			field := t.Field(i)
+			value := v.Field(i)
+			kind := value.Kind()
+			if kind == reflect.Struct || kind == reflect.Slice || kind == reflect.Map {
+				if value.CanInterface() {
+					if value.Type().String() == "time.Time" {
+						setInMap(result, field, value.Interface())
+						break
```
Comment (wongoo) [inline filter/filter_impl/generic_filter.go:103]: should be `continue`

File: filter/filter_impl/generic_filter.go [hunk 3]
```diff
@@ -93,13 +93,20 @@ func struct2MapAll(obj interface{}) interface{} {
...
 		for i := 0; i < t.NumField(); i++ {
-			if v.Field(i).Kind() == reflect.Struct || v.Field(i).Kind() == reflect.Slice || v.Field(i).Kind() == reflect.Map {
-				if v.Field(i).CanInterface() {
-					setInMap(result, t.Field(i), struct2MapAll(v.Field(i).Interface()))
+			field := t.Field(i)
+			value := v.Field(i)
+			kind := value.Kind()
+			if kind == reflect.Struct || kind == reflect.Slice || kind == reflect.Map {
+				if value.CanInterface() {
+					if value.Type().String() == "time.Time" {
```
Comment (wongoo) [inline filter/filter_impl/generic_filter.go:101]: use a type check  expression would be better: `switch v := param.(type)`

**Review by wongoo — APPROVED** [review]

**Pull request merged**